lib/Sema/CSSimplify.cpp

```diff
@@ -4774,6 +4775,25 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
                                       flags, locatorB);
     };
 
+    // Check if any property wrappers on the base of the member lookup have
+    // mactching members that we can fall back to.
+    if (auto dotExpr =
+            dyn_cast_or_null<UnresolvedDotExpr>(locator->getAnchor())) {
+      auto baseExpr = dotExpr->getBase();
+      auto resolvedOverload = findSelectedOverloadFor(baseExpr);
+      if (auto wrappedProperty =
```

Comment:
If the property wrapper type has a `wrapperValue` property (described [here](https://github.com/apple/swift-evolution/blob/master/proposals/0258-property-wrappers.md#delegating-access-to-the-storage-property)), then `$foo` will have the type of the `wrapperValue` property (with appropriate substitutions applied), not the backing storage type.